packages/@react-spectrum/s2/src/SelectBox.tsx

```diff
@@ -0,0 +1,304 @@
+    }
+  }
+}, getAllowedOverrides());
+
+const iconContainer = style({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  size: {
+    XS: 16,
+    S: 20,
+    M: 24,
+    L: 28,
+    XL: 32
+  },
+  flexShrink: 0,
+  color: {
+    isDisabled: 'disabled'
+  }
+}, getAllowedOverrides());
+
+const textContainer = style({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: {
+    default: 'center',
+    orientation: {
+      horizontal: 'start'
+    }
+  },
+  gap: 'text-to-visual',
+  color: {
+    isDisabled: 'disabled'
+  }
+}, getAllowedOverrides());
+
+const descriptionText = style({
+  display: {
+    default: 'none',
+    orientation: {
+      horizontal: 'block'
+    }
+  },
+  font: 'ui-sm',
+  color: {
+    default: 'gray-600',
+    isDisabled: 'disabled'
+  },
+  lineHeight: 'body'
+}, getAllowedOverrides());
+
+const checkboxContainer = style({
+  position: 'absolute',
+  top: 16,
+  left: 16
+}, getAllowedOverrides());
+
+/**
+ * SelectBox components allow users to select options from a list.
+ * They can behave as radio buttons (single selection) or checkboxes (multiple selection).
+ */
+export const SelectBox = /*#__PURE__*/ forwardRef(function SelectBox(props: SelectBoxProps, ref: FocusableRef<HTMLLabelElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, SelectBoxItemContext);
+  let {children, value, isDisabled = false, isSelected, onChange, UNSAFE_className = '', UNSAFE_style} = props;
+  let inputRef = useRef<HTMLInputElement | null>(null);
+  let domRef = useFocusableRef(ref, inputRef);
+  
+  let {
+    allowMultiSelect = false,
+    size = 'M',
+    orientation = 'vertical'
+  } = useContext(SelectBoxContext);
+
+  const Selector = allowMultiSelect ? AriaCheckbox : AriaRadio;
+  
+  return (
+    <Selector
+      value={value}
+      isDisabled={isDisabled}
+      isSelected={isSelected}
+      onChange={() => onChange?.(isSelected ?? false)}
+      ref={domRef}
+      inputRef={inputRef}
+      style={UNSAFE_style}
+      className={renderProps => UNSAFE_className + selectBoxStyles({...renderProps, size, orientation}, props.styles)}>
+      {renderProps => {
+        // Separate icon and text content from children
+        const childrenArray = React.Children.toArray(children);
+        const iconElement = childrenArray.find((child: any) => child?.props?.slot === 'icon');
+        const textElement = childrenArray.find((child: any) => child?.props?.slot === 'text');
+        const descriptionElement = childrenArray.find((child: any) => child?.props?.slot === 'description');
+        const otherChildren = childrenArray.filter((child: any) => 
+          !['icon', 'text', 'description'].includes(child?.props?.slot)
+        );
+
+        return (
+          <>
+            {(renderProps.isSelected || renderProps.isHovered || renderProps.isFocusVisible) && (
+              <div className={checkboxContainer({...renderProps, size}, props.styles)}>
+                <Checkbox 
```

Comment:
one thing i noticed is that if i have `selectionMode='single'`, hover over a SelectBox, and click on the checkbox, nothing happens. however, i would expect that the checkbox be marked as selected. 

does this need to be a CheckBox component or can we just style something to look like a checkbox? i don't think having nested label elements is valid. it might cause issues with forms. but also right now, i can keyboard navigate to both the select box and the checkbox inside of it. i'm not sure if that's expected or not